app/lib/actions/pulp/repository/distributor_publish.rb

```diff
@@ -5,7 +5,7 @@ class DistributorPublish < Pulp::AbstractAsyncTask
         middleware.use Actions::Middleware::SkipIfMatchingContent
 
         def plan(repository, smart_proxy, options)
-          options = {:smart_proxy_id => smart_proxy.id, :options => options, :dependency => options[:dependency]}
+          options = {:smart_proxy_id => smart_proxy.id, :options => options, :dependency => options[:dependency], :matching_content => options[:matching_content]}
```

Comment:
this is too late. I think we should change https://github.com/Katello/katello/blob/master/app/lib/actions/middleware/skip_if_matching_content.rb#L15 instead  and have him check something like
```
        options = action.input[:options]
        if options && options.keys.include?('matching_content') && options['matching_content']
```